From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: GliczDev <67753196+GliczDev@users.noreply.github.com>
Date: Mon, 6 Jul 2026 21:15:03 +0200
Subject: [PATCH] Unwrap CaptureType to WildcardType in
 TypeFactory#checkParametersWithinBound


diff --git a/src/main/java/io/leangen/geantyref/TypeFactory.java b/src/main/java/io/leangen/geantyref/TypeFactory.java
index 43dd5d79422a214d12ddb5a75e88a71f2d7793c4..d5b97eca47276cfa3cefeefffe1cea5a927fe467 100644
--- a/src/main/java/io/leangen/geantyref/TypeFactory.java
+++ b/src/main/java/io/leangen/geantyref/TypeFactory.java
@@ -189,31 +189,36 @@ public class TypeFactory {
 
         // for every bound on every parameter
         for (int i = 0; i < arguments.length; i++) {
+            Type argument = arguments[i];
+            if (argument instanceof CaptureType) {
+                argument = ((CaptureType) argument).getWildcardType();
+            }
+
             for (Type bound : typeParameters[i].getBounds()) {
                 // replace type variables in the bound by their value
                 Type replacedBound = varMap.map(bound);
 
 
-                if (arguments[i] instanceof WildcardType) {
-                    WildcardType wildcardTypeParameter = (WildcardType) arguments[i];
+                if (argument instanceof WildcardType) {
+                    WildcardType wildcardTypeParameter = (WildcardType) argument;
 
                     // Check if a type satisfying both the bounds of the variable and of the wildcard could exist
 
                     // upper bounds must not be mutually exclusive
                     for (Type wildcardUpperBound : wildcardTypeParameter.getUpperBounds()) {
                         if (!couldHaveCommonSubtype(replacedBound, wildcardUpperBound)) {
-                            throw new TypeArgumentNotInBoundException(arguments[i], typeParameters[i], bound);
+                            throw new TypeArgumentNotInBoundException(argument, typeParameters[i], bound);
                         }
                     }
                     // a lowerbound in the wildcard must satisfy every upperbound
                     for (Type wildcardLowerBound : wildcardTypeParameter.getLowerBounds()) {
                         if (!GenericTypeReflector.isSuperType(replacedBound, wildcardLowerBound)) {
-                            throw new TypeArgumentNotInBoundException(arguments[i], typeParameters[i], bound);
+                            throw new TypeArgumentNotInBoundException(argument, typeParameters[i], bound);
                         }
                     }
                 } else {
-                    if (!GenericTypeReflector.isSuperType(replacedBound, arguments[i])) {
-                        throw new TypeArgumentNotInBoundException(arguments[i], typeParameters[i], bound);
+                    if (!GenericTypeReflector.isSuperType(replacedBound, argument)) {
+                        throw new TypeArgumentNotInBoundException(argument, typeParameters[i], bound);
                     }
                 }
             }
